cmd/posix.go

```diff
@@ -72,6 +72,51 @@ func checkPathLength(pathName string) error {
 	return nil
 }
 
+func checkPathValid(path string) (string, error) {
+	if path == "" {
+		return path, errInvalidArgument
+	}
+
+	var err error
+	// Disallow relative paths, figure out absolute paths.
+	path, err = filepath.Abs(path)
+	if err != nil {
+		return path, err
+	}
+
+	fi, err := os.Stat(path)
+	if err != nil && !os.IsNotExist(err) {
+		return path, err
+	}
+	if os.IsNotExist(err) {
+		// Disk not found create it.
+		if err = os.MkdirAll(path, 0777); err != nil {
+			return path, err
+		}
+	}
+	if fi != nil && !fi.IsDir() {
+		return path, syscall.ENOTDIR
+	}
+
+	di, err := getDiskInfo(path)
+	if err != nil {
+		return path, err
+	}
+	if err = checkDiskMinTotal(di); err != nil {
+		return path, err
+	}
+
+	// check if backend is writable.
+	file, err := os.Create(pathJoin(path, ".writable"))
+	if err != nil {
+		return path, err
+	}
+	file.Close()
+
+	err = os.Remove(pathJoin(path, ".writable"))
```

Comment:
I think it would be better to name this temporary file as `.writable-check.tmp`.   This way a stale `.writable` file would be removed by admin if needed.